### PR TITLE
UI improvements

### LINKED ISSUE #11: Front page
**Description of new feature**

_What should the new feature do?  For visual features, include an image/mockup of the expected output._

- Build a front page that allows the user to submit a query.
- Incl. an upload button.  After the file is selected, the query is submitted.
- Show a loading spinner, that replaces the upload button, when the query is submitted.
- Error message on file error.

Depends on:

- #12 
- #13 
- #8 

**Motivation for new feature**

_Describe your use case for this new feature._

Provide a place for the user to start a query.

### LINKED ISSUE #15: Initial network layout
**Description of new feature**

_What should the new feature do?  For visual features, include an image/mockup of the expected output._

Run a layout (probably on the client) to organise the network.

- Interesting possible layouts
  - [FCOSE](https://github.com/iVis-at-Bilkent/cytoscape.js-fcose) -- super fast FD
  - [CISE](https://github.com/iVis-at-Bilkent/cytoscape.js-cise) -- can make create use of clusters (if nodes have a cluster ID defined)
- It doesn't have to be perfect for the MVP, just 'good enough'. 

**Motivation for new feature**

_Describe your use case for this new feature._

We need to lay out the nodes somehow.

### LINKED ISSUE #20: Initial network style
**Description of new feature**

_What should the new feature do?  For visual features, include an image/mockup of the expected output._

Give the network a good stylesheet.

- Fixed stylesheet for the MVP.  No user customisation
- Node size mapped to 'importance' -- not pathway size.  NES
- Labels shouldn't overtake the figure.  Use features like text wrapping, font sizing, or cutoffs (with ellipses).
- Colours should match with sidebar gene bars where the same metric/score is represented.

**Motivation for new feature**

_Describe your use case for this new feature._

The network needs some simple, initial style for display.

### PR DESCRIPTION
**Pull request by maxkfranz**
**General information**

Associated issues:

- Closes #11 
- Closes #20 
- Closes #15

**Checklist**

Author:

- [x] One or more reviewers have been assigned.
- [x] Automated tests have been included in this pull request, if possible, for the new feature(s) or bug fix. -- N/A (UI)
- [x] The associated GitHub issues are included (above).
- [x] Notes have been included (below).

Reviewers:

- [ ] All automated checks are passing (green check next to latest commit).
- [ ] At least one reviewer has signed off on the pull request.  Reviewers have two business days to review the pull request, after which the author may merge in the pull request unilaterally.

**Notes**

Improve usability of the front page:

- The entire page is a drag-and-drop file area.  Drag a file anywhere in the page and the file will be uploaded for your query.
- Clearer tag line:  "Get a quick-and-easy enrichment figure for your two-case RNA-Seq experiment."
- Nice, big, blue upload button:  Centred in the page so it's the most salient thing.  Button text now reads "Enrich my RNA-Seq data".  UI should be context sensitive to the user's use case, not telling them what technical step to do (e.g. "Click here to upload file").
- Better error message:  The error message is based on the user's perspective, not the developer's perspective.  We show a human-understandable error message so that the user knows what to do.  We don't show technical error messages that the user can't act on.
- The spinner and error message are centred, like the upload button.
- There's a debug menu for selecting canned input files.  To add files to the list, just add a file to the `sample-data` directory.

https://user-images.githubusercontent.com/989043/203342292-37f14b9f-f9d4-42a7-938d-5f531fc2757d.mp4

Update network style & layout:

- Thicker edges
- Regular COSE instead of FCOSE.  Edge weighting.
- Layout always runs via controller.
- Manual (non-COSE) packing of disconnected (zero degree) nodes via grid layout.

Screen Shot 2022-11-22 at 09 41 37

Misc. changes:

- Text updates
- Merges in with Mike's initial expression file support.  Should work with debug menu.  May need better handling of canned expression data re. classes in future.

### CODE REVIEW

File: public/sample-data/mesenvs-immuno-xsm.rnk [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+NDUFV2	-4.73097513116836
+DDIT3	-6.20075956818443
+DDX60	-7.23165518082732
+GBP1	-8.77775627864623
+NUDT16P1	-9.36142904483035
+CALN1	-10.7079701444496
+ATPAF1	-12.7812820237903
+TAP2	-14.8247058519209
+CSAG3	-16.7307784891732
+PSMB9	-18.3590574351204
```
Comment (maxkfranz) [inline public/sample-data/mesenvs-immuno-xsm.rnk:27]: @chrtannus, this is your XSM sample file with both positive and negative values.

**Comment by mikekucera** [comment]
@maxkfranz Are you planning to add more to the front page in the future, or do you expect the final result to be this minimalistic?

**Comment by maxkfranz** [comment]
There are several things to consider in future. Do you have something particular in mind?

**Comment by maxkfranz** [comment]
Great.  You’ve identified a number of landing page items.  Let’s track an issue for scoping it out.  For instance, there may be other important things to include (e.g. linking to the app note and citation information).

Re. this PR, have you run into any bugs?

**Comment by chrtannus** [comment]
I haven't noticed any issues. It looks good to me.

**Comment by mikekucera** [comment]
This PR seems fine wrt bugs.

The validation of the uploaded file needs to be more robust, and the UI for selecting classes needs to be improved, but those can be separate issues.

**Review by mikekucera — APPROVED** [review]

**Review by chrtannus — APPROVED** [review]

**Comment by maxkfranz** [comment]
> The validation of the uploaded file needs to be more robust, and the UI for selecting classes needs to be improved, but those can be separate issues.

See new issues:

- Refined class selection UI #41
- Improved validation of input files #42

They're just stubbed out, so feel free to edit the original posts or add comments to scope things out.

**Pull request merged**